src/benchmark/file_io_write_mmap_benchmark.cpp

```diff
@@ -84,6 +89,113 @@ void FileIOWriteMmapBenchmarkFixture::mmap_write_benchmark(benchmark::State& sta
   close(fd);
 }
 
+void write_mmap_chunk_sequential(const size_t from, const size_t to, int32_t* map, uint32_t* data_to_write_start) {
+  const auto uint32_t_size = ssize_t{sizeof(uint32_t)};
+  const auto bytes_to_write = static_cast<ssize_t>(uint32_t_size * (to - from));
+  memcpy(map + from, data_to_write_start + from, bytes_to_write);
+}
+
+void write_mmap_chunk_random(const size_t from, const size_t to, int32_t* map,
+                             const std::vector<uint32_t>& data_to_write,
+                             const std::vector<uint32_t>& ind_access_order) {
+  for (auto idx = from; idx < to; ++idx) {
+    auto access_index = ind_access_order[idx];
+    map[access_index] = data_to_write[idx];
+  }
+}
+
+void FileIOWriteMmapBenchmarkFixture::mmap_write_multi_threaded(benchmark::State& state, const int mmap_mode_flag,
+                                                                const int data_access_mode, const ssize_t file_size,
+                                                                uint16_t thread_count) {
+  auto fd = int32_t{};
+  Assert(((fd = open(filename, O_RDWR)) >= 0), fail_and_close_file(fd, "Open error:", errno));
+
+  //set output file size to avoid mapping errors
+  Assert((ftruncate(fd, NUMBER_OF_BYTES) == 0), fail_and_close_file(fd, "Ftruncate error:", errno));
+
+  auto threads = std::vector<std::thread>(thread_count);
+  auto batch_size = static_cast<uint64_t>(std::ceil(static_cast<float>(NUMBER_OF_ELEMENTS) / thread_count));
+
+  for (auto _ : state) {
+    state.PauseTiming();
+    micro_benchmark_clear_disk_cache();
+    state.ResumeTiming();
+
+    // Getting the mapping to memory.
+    const auto OFFSET = off_t{0};
+    auto* map = reinterpret_cast<int32_t*>(mmap(NULL, NUMBER_OF_BYTES, PROT_WRITE, mmap_mode_flag, fd, OFFSET));
+    Assert(map != MAP_FAILED, "Mapping Failed:" + std::strerror(errno));
+
+    auto* data_to_write_start = std::data(data_to_write);
+
+    switch (data_access_mode) {
+      case SEQUENTIAL:
+        for (auto i = size_t{0}; i < thread_count; i++) {
+          auto from = batch_size * i;
+          auto to = from + batch_size;
+          if (to >= NUMBER_OF_ELEMENTS) {
+            to = NUMBER_OF_ELEMENTS;
+          }
```

Comment:
This suggestion  reduces code lines, but arguably increases complexity. What do you think?
```suggestion
          auto to = min(from+batch_size, NUMBER_OF_ELEMENTS);
```

Reply:
could be applied below aswell

Reply:
i actually like it, i think it increases readability